source/math/FloatComplexVectorList.ooc

```diff
@@ -30,21 +30,26 @@ FloatComplexVectorList: class extends VectorList<FloatComplex> {
 		super(other _vector)
 		this _count = other count
 	}
+	init: func ~withValue (capacity: Int, value: FloatComplex) {
+		super(capacity)
+		for (i in 0 .. capacity)
+			this add(value)
+	}
 	toVectorList: func -> VectorList<FloatComplex> {
 		result := VectorList<FloatComplex> new()
 		result _vector = this _vector
 		result _count = this _count
 		result
 	}
-	sum: func -> FloatComplex {
-		result := FloatComplex new()
-		for (i in 0 .. this _count)
-			result = result + this[i]
-		result
-	}
-	mean: func -> FloatComplex {
-		sum() / this _count
+	sum: FloatComplex {
+		get {
+			result := FloatComplex new()
+			for (i in 0 .. this _count)
+				result = result + this[i]
+			result
+		}
 	}
+	mean ::= this sum / this _count
```

Comment:
I still don't understand it. From this guide (thanks @davidhesselbom ) https://msdn.microsoft.com/en-us/library/vstudio/ms229054%28v=vs.100%29.aspx : "*properties should not be computationally complex or produce side effects.*"
This `mean` property definitely does not follow this guideline - it is computationally complex. Later when this property is used somewhere it won't be clear that it calls a complex function behind the scenes, like
```ooc
for (i in 0 .. bignumber)
    dosomething(vector mean)
```
compared to 
```ooc
for (i in 0 .. bignumber)
    dosomething(vector mean()) // now I see you, you sneaky little funtction...
```
it looks like `vector mean` is a variable, while it is just a function lurking behind obfuscated syntax. In order to find out, one needs to open the definition of this class. So right now it looks to me like our guidelines about properties are : "everything that is read-only should be a property". It should be more like "everything that is read-only and cheap to compute should be a property".
That's why I don't get it. Something that looks like a variable access but actually calls several functions every time it is used is just unnecessary confusion.